wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerHandler.java

```diff
@@ -96,28 +84,37 @@ protected boolean shouldCopyHeader(@Nullable NamespacedId workspaceId) {
     }
   }
 
+  protected Connection getConnection(NamespacedId connectionId) {
+    return TransactionRunners.run(getContext(), context -> {
+      ConnectionStore store = ConnectionStore.get(context);
+      return store.get(connectionId);
+    }, ConnectionNotFoundException.class);
+  }
+
+  protected Connection getValidatedConnection(NamespacedId connectionId, ConnectionType expectedType) {
+    return TransactionRunners.run(getContext(), context -> {
+      ConnectionStore store = ConnectionStore.get(context);
+      return getValidatedConnection(store, connectionId, expectedType);
+    }, ConnectionNotFoundException.class);
```

Comment:
should this also include `IOException` ?

Reply:
Also what would happen if run method throws a RuntimeException `BadRequestException`?

Reply:
hm, actually I don't need to explicitly mention any of these exceptions since they're all RuntimeException, which gets propagated by TransactionsRunners. Will change these.